packages/mcp-provider-devops/src/shared/auth.ts

```diff
@@ -0,0 +1,151 @@
+    }
+
+    return taggedOrgs;
+  }
+
+
+
+// This function is the main entry point for Tools to get an allowlisted Connection
+export async function getConnection(username: string): Promise<Connection> {
+    // We get all allowed orgs each call in case the directory has changed (default configs)
+    const allOrgs = await getAllAllowedOrgs();
+    const foundOrg = findOrgByUsernameOrAlias(allOrgs, username);
+  
+    if (!foundOrg)
+      return Promise.reject(
+        new Error(
+          'No org found with the provided username/alias. Ask the user to specify one or check their MCP Server startup config.'
+        )
+      );
+  
+    const authInfo = await AuthInfo.create({ username: foundOrg.username });
+    const connection = await Connection.create({ authInfo });
+    return connection;
+  }
+
+
+  export function findOrgByUsernameOrAlias(
+    allOrgs: SanitizedOrgAuthorization[],
+    usernameOrAlias: string
+  ): SanitizedOrgAuthorization | undefined {
+    return allOrgs.find((org) => {
+      // Check if the org's username or alias matches the provided usernameOrAlias
+      const isMatchingUsername = org.username === usernameOrAlias;
+      const isMatchingAlias = org.aliases && Array.isArray(org.aliases) && org.aliases.includes(usernameOrAlias);
+  
+      return isMatchingUsername || isMatchingAlias;
+    });
+  }
+
+  /**
+   * Get only the DevOps Center org for commit operations
+   */
+  export async function getDoceHubOrg(): Promise<{
+    doceHub: (SanitizedOrgAuthorization & { orgType?: string }) | null;
+    error?: string;
+  }> {
+    const allOrgs = await getAllAllowedOrgs();
+    
+    // Find DevOps Center org
+    const doceHub = allOrgs.find(org => org.orgType === 'DevOps Center') || null;
+
+    let error = '';
+    if (!doceHub) {
+      error += 'DevOps Center org not found. ';
+    }
+
+    return {
+      doceHub,
+      error: error || undefined
+    };
+  }
+
+  /**
+   * Get both DevOps Center and Sandbox orgs with validation
+   */
+  export async function getRequiredOrgs(devopsUsername: string, sandboxUsername: string ): Promise<{
+    doceHub: (SanitizedOrgAuthorization & { orgType?: string }) | null;
+    sandbox: (SanitizedOrgAuthorization & { orgType?: string }) | null;
+    error?: string;
+  }> {
+    const allOrgs = await getAllAllowedOrgs();
+
+    // Resolve from available orgs strictly by username
+    const doceHub = allOrgs.find(org => org.username === devopsUsername) || null;
+    const sandbox = allOrgs.find(org => org.username === sandboxUsername) || null;
+
+    let error = '';
+    if (!doceHub) {
+      error += `DevOps Center org '${devopsUsername}' not found. Login with sf auth:web:login or set Devops_org_username. `;
+    }
+    if (!sandbox) {
+      error += `Sandbox org '${sandboxUsername}' not found. Login with sf auth:web:login or set Sandbox_org_username. `;
```

Comment:
> Devops_org_username Sandbox_org_username

are these referring to env vars? it's not clear from the msg
The CLI command is `sf org login web`.

fwiw, A4D locks the MCP config so your users wouldn't be able to set these env vars by themselves.